### ISSUE TITLE
[change] Invert upload/download on interfaces that are member of a bridge

### ISSUE THREAD

**Issue by nemesisdesign** [openwisp/lua-monitoring#22]
In Linux, bridged interface show tx/rx packets in reverse: download is upload and upload is download.

I couldn't find a canonical reference on why this is the way it is, if anyone finds it, let me know.

Now the problem we have is that in the traffic graph, the upload is greater than the download on wifi interfaces that are part of a bridge, which most users will consider as a bug.

![Screenshot from 2021-01-10 18-01-07](https://user-images.githubusercontent.com/841044/104137853-ded56600-536d-11eb-83e4-41e23ab226f0.png)

So I think we should do the following:

- whenever an interface is flagged as an interface which should be traffic monitored, we should check whether this interface is part of a bridge
- if it is, we should simply invert rx and tx

**Addendum**: I'm trying to do some more research on the subject before proceeding with this: https://forum.openwrt.org/t/inverted-tx-rx-packets-in-bridged-interfaces/85093.

**Comment by nemesisdesign** [comment]
**Addendum**: I'm trying to do some more research on the subject before proceeding with this: https://forum.openwrt.org/t/inverted-tx-rx-packets-in-bridged-interfaces/85093.

**Comment by nemesisdesign** [comment]
During testing I found out my assumption was wrong.
I think I understood now: is not that rx/tx are inverted, is that the interface tx/rx statistics of a wifi access point interface installed on a router have a different perspective: for the wifi interface, every packet that the user downloads is **transmitted** (tx) to the user, while every packet that the user uploads is **received** (rx) by the interface, that explains why tx bytes is greater than rx bytes on these interfaces.
I do not think it has to do anything with the fact that the interface is bridged: a tap (layer2 VPN) member of a bridge which is used to tunnel traffic still shows received packets greater than sent packets, so that's why I changed the code further in https://github.com/openwisp/lua-monitoring/commit/266d98a4ec6eebdede227a75509eb4d11d4735cc to ensure the inversion is done only on wifi acess points and I made clear in the comment that we do this to show data from the perspective of the user rather than the perspective of the user.
